components/raftstore/src/store/peer.rs

```diff
@@ -2233,18 +2235,29 @@ impl Peer {
         req: &mut RaftCmdRequest,
     ) -> Result<()> {
         let last_index = self.raft_group.raft.raft_log.last_index();
-        let min_progress = self.get_min_progress()?;
-        let min_index = min_progress + 1;
-        if min_progress == 0 || last_index - min_progress > ctx.cfg.merge_max_log_gap {
+        let (min_matched, min_committed) = self.get_min_progress()?;
```

Comment:
The correctness depends on `min_matched` must be greater than or equal to `min_committed`.
Although it is right, I think it's better to use assert or panic to defend some unexpected fault.